python_requires

```diff
@@ -18,32 +18,37 @@ import argparse
 import glob
 import os
 import re
-import tarfile
 
+from metaextract import utils as meta_utils
 from packaging.requirements import Requirement
 
 
-def read_from_tarball(tar_name, filename):
-    with tarfile.open(tar_name) as tar:
-        try:
-            file_found = min((f for f in tar.getnames()
-                              if f.endswith(filename)),
-                             key=len)
-        except ValueError:
-            raise Exception("Coud not find a file %s in %s" %
-                            (filename, tar_name))
-        t = tar.getmember(file_found)
-        file_content = tar.extractfile(t).read()
+def _requires_get_source(all_requires, pkg):
+    if pkg in all_requires['install']:
+        return 'install'
+    elif pkg in all_requires['extras']:
+        return 'extras'
+    elif pkg in all_requires['tests']:
+        return 'tests'
+    else:
+        return 'unknown'
+
 
-    return file_content
+def _get_complete_requires(all_requires):
+    """get a dict with all requirements from 'install', 'extras' and 'tests'"""
+    complete_requires = dict()
+    complete_requires.update(all_requires['install'])
+    complete_requires.update(all_requires['extras'])
+    complete_requires.update(all_requires['tests'])
```

Comment:
Not sure here, but this order will make more important 'tests' that 'install'. So 'tests' can overwrite 'install' and 'extras'. Maybe use the reverse order for updating?